cloud/ec2_price.go

```diff
@@ -30,21 +30,6 @@ type cachingPriceFetcher struct {
 
 var pkgCachingPriceFetcher *cachingPriceFetcher
 
-func getPkgCachingPriceFetcher() *cachingPriceFetcher {
-	if pkgCachingPriceFetcher == nil {
-		pkgCachingPriceFetcher = new(cachingPriceFetcher)
-	}
-	pkgCachingPriceFetcher.Lock()
-	return pkgCachingPriceFetcher
-}
-
-func putPkgCachingPriceFetcher(fetcher *cachingPriceFetcher) {
-	defer fetcher.Unlock()
-	if fetcher != pkgCachingPriceFetcher {
-		grip.EmergencyPanic("fetcher passed to putPkgCachingPriceFetcher is not package fetcher")
-	}
-}
```

Comment:
I think having a get method is fine, as long as you do the new in a package init()

I didn't like the panic and the put, but I think the get is fine

Reply:
I don't think they're necessary at all, actually, after moving to an init function. I simplified the logic.